src/angularjs/LineChart.ts

```diff
@@ -105,7 +105,7 @@ module n3Charts {
       };
 
       scope.$watch('options', updateAll, true);
-      scope.$watch('data', updateData, scope.options.deepWatchData);
+      scope.$watch('data', updateData, options ? options.deepWatchData : true);
```

Comment:
`options` is set to an `Options.Options` object constructed from `scope.options` by the `updateAll` method, which is run by the watch on `scope.options`.  So after that watch runs, `options.deepWatchData` will be the same as `scope.options.deepWatchData`, defaulting to `true` if the latter doesn't specify it.

But it's possible the `LineChart` can be initialized with an empty `scope.options` and then set the latter later.  The conditional here catches that case and prevents it from throwing an exception because `scope.options` and `options` are undefined.